builder/frameworks/espidf.py

```
@@ -102,38 +108,32 @@ def create_silent_action(action_func):
-    assert ARDUINO_FRAMEWORK_DIR and os.path.isdir(ARDUINO_FRAMEWORK_DIR)
-    arduino_libs_mcu = join(ARDUINO_FRAMEWORK_DIR,"tools","esp32-arduino-libs",mcu)
+    arduino_pkg_dir = Path(platform.get_package_dir("framework-arduinoespressif32"))
+    # Rename using the symlink path (do not resolve before rename)
+    if "@" in arduino_pkg_dir.name:
+        new_dir = arduino_pkg_dir.with_name(arduino_pkg_dir.name.replace("@", "-"))
+        os.rename(str(arduino_pkg_dir), str(new_dir))
+        arduino_pkg_dir = new_dir
+    ARDUINO_FRAMEWORK_DIR_PATH = arduino_pkg_dir.resolve()
+    ARDUINO_FRAMEWORK_DIR = str(ARDUINO_FRAMEWORK_DIR_PATH)
+    if not ARDUINO_FRAMEWORK_DIR or not os.path.isdir(ARDUINO_FRAMEWORK_DIR):
+        raise RuntimeError(f"Arduino framework directory not found: {ARDUINO_FRAMEWORK_DIR}")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)